src/js/angular/similarity/controllers/similarity-list.controller.js

```diff
@@ -59,26 +72,26 @@ function SimilarityCtrl($scope, $interval, toastr, $repositories, $licenseServic
         }
     };
 
-    $scope.goToSimilarityIndex = function (index) {
-        if (!('BUILT' === index.status || 'OUTDATED' === index.status || 'REBUILDING' === index.status)) {
+    $scope.goToSimilarityIndex = (index) => {
+        if (!(SimilarityIndexStatus.BUILT === index.status || SimilarityIndexStatus.OUTDATED === index.status || SimilarityIndexStatus.REBUILDING === index.status)) {
             return;
         }
         $scope.empty = true;
         if ($scope.selected !== index) {
             $scope.lastSearch = undefined;
             $scope.selected = index;
         }
-        if (index.type === 'text') {
-            $scope.searchType = 'searchTerm';
-        } else if (index.type === 'predication') {
-            $scope.searchType = 'searchEntity';
+        if (SimilarityIndexType.TEXT === index.type) {
+            $scope.searchType = SimilaritySearchType.SEARCH_TERM;
+        } else if (SimilarityIndexType.PREDICATION === index.type) {
+            $scope.searchType = SimilaritySearchType.SEARCH_ENTITY;
         }
-        if (index.type === 'text' || index.type === 'predication') {
+        if (SimilarityIndexType.TEXT === index.type || SimilarityIndexType.PREDICATION === index.type) {
```

Comment:
I think that it would be more convenient if we have some utility functions for all these checks either in the model or here in the controller.

Reply:
Yes, I have it in mind. I will do it. First I want to extract the types and then will continue with this.